core/src/main/java/com/yahoo/imapnio/async/data/MessageNumberSet.java

```diff
@@ -189,7 +189,7 @@ public boolean equals(final Object obj) {
             v.add(new MessageNumberSet(start, end));
             i = j;
         }
-        return v.toArray(new MessageNumberSet[v.size()]);
```

Comment:
I think original code is better, because we allocate exactly memory size for MessageNumberSet array based on v.size(). If you changed to 0, it needs to allocate default size in toArray(), which is not as memory efficient as before.

Reply:
Looks good now